lib/plugins/aws/package/compile/events/schedule.js

```diff
@@ -68,16 +69,60 @@ class AwsCompileScheduledEvents {
               },
               required: ['inputTemplate'],
               additionalProperties: false,
+            }
+          },
+          required: ['rate'],
+          additionalProperties: false,
+        },
+        {
+          type: 'object',
+          properties: {
+            rate: {
+              type: 'array',
+              minItems: 1,
+              items: {
+                type: 'string',
+                pattern: scheduleSyntax,
+              },
+            },
+            enabled: { type: 'boolean' },
+            name: { type: 'string', minLength: 1, maxLength: 64, pattern: '[\\.\\-_A-Za-z0-9]+' },
+            description: { type: 'string', maxLength: 512 },
+            input: {
+              anyOf: [
+                { type: 'string', maxLength: 8192 },
+                {
+                  type: 'object',
+                  oneOf: [
+                    {
+                      properties: {
+                        body: { type: 'string', maxLength: 8192 },
+                      },
+                      required: ['body'],
+                      additionalProperties: false,
+                    },
+                    {
+                      not: {
+                        required: ['body'],
+                      },
+                    },
+                  ],
+                },
+              ],
             },
             method: {
               type: 'string',
-              enum: ['eventBus', 'scheduler'],
+              enum: ['eventBus'],
             },
             roleArn: {
               type: 'string',
             },
+            timezone: {
+              type: 'string',
+              enum: TIME_ZONES
+            }
           },
-          required: ['rate'],
+          required: ['rate', 'method'],
```

Comment:
Method doesn't need to be required since there's a default. Please revert

Reply:
To explain what I've done with the json-schema. I've created two different schemas for schedule, one for eventbus schedule which stays as the old schema and a new one for scheduler which adds new properties -> method, roleArn, timezone and removes inputPath and inputTransformer.
Doing it this way will ensure that people don't mix the two and schema check will detect and let them know that these properties are exclusive and cannot be used together, this is why method is required in the second schema.
I've added title to prevent confusion.

Reply:
This would introduce a breaking change to the API, which is not what we want here. I understand what's going on, but it should be non-breaking, which is why I suggested a revert and to go back to the original properties format.

Reply:
I'm sorry, why would it break the api?